File: 04_popView/04_popView/ViewController.m
//
//  ViewController.m
//  04_popView
//
//  Created by moxiaoyan on 2019/5/26.
//  Copyright © 2019 moxiaoyan. All rights reserved.
//

#import "ViewController.h"
#import "PopViewController.h"

@interface ViewController () <UIAdaptivePresentationControllerDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
  btn.backgroundColor = UIColor.redColor;
  [btn addTarget:self action:@selector(click) forControlEvents:UIControlEventTouchUpInside];
  [self.view addSubview:btn];
  btn.frame = CGRectMake(50, 50, 50, 50);
}

- (void)click {
  PopViewController *vc = [[PopViewController alloc] init];
  vc.modalPresentationStyle = UIModalPresentationFormSheet;
  vc.preferredContentSize = CGSizeMake(100, 100);
  vc.presentationController.delegate = self;
  [self presentViewController:vc animated:YES completion:nil];
}

#pragma mark - UIAdaptivePresentationControllerDelegate
- (UIModalPresentationStyle)adaptivePresentationStyleForPresentationController:(UIPresentationController *)controller {
  return UIModalPresentationNone;
}

@end

File: 07_QRScan/07_QRScan/ViewController.swift
//
//  ViewController.swift
//  07_QRScan
//
//  Created by moxiaoyan on 2019/7/11.
//  Copyright © 2019 moxiaoyan. All rights reserved.
//

import UIKit
import AVFoundation

class ViewController: UIViewController {
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    let btn = UIButton(type: .custom)
    btn.frame = CGRect(x: 0, y: 0, width: 120, height: 50)
    btn.center = self.view.center
    btn.setTitle("扫描二维码", for: .normal)
    btn.setTitleColor(.red, for: .normal)
    btn.addTarget(self, action: #selector(clickScan), for: .touchUpInside)
    self.view.addSubview(btn)
    
  }
  
  @objc func clickScan() {
    let nav = UINavigationController(rootViewController: MOQRScanViewController())
    self.present(nav, animated: true, completion: nil)
  }

}
